### PR TITLE
Set "typeofs: false" to prevent mangling typeofs by uglify

### PR DESCRIPTION
**Pull request by lxe**
`typeofs: true (default)` transforms typeof foo == "undefined" into foo === void 0.

This mangles mapbox-gl creating an error when used alongside with window global mangling:

https://github.com/webpack-contrib/uglifyjs-webpack-plugin/issues/189

### CODE REVIEW

**Comment by lxe** [comment]
cc @vicapow

**Comment by vicapow** [comment]
Do you have other compile configuration tests we can use as an example?

**Comment by KevinGrandon** [comment]
> Do you have other compile configuration tests we can use as an example?

It depends on the behavior of what happens if there was an error. It sounds like this possibly exhibits as a node.js runtime error? If we can add a fixture with logic that reproduces this to the [fixtures directory](https://github.com/fusionjs/fusion-cli/tree/master/test/fixtures), we should be able to add a test like the following which ensures SSR works: https://github.com/fusionjs/fusion-cli/blob/9e872397fc1fe2df4025211d2158ca05072f9656/test/cli/build.js#L444-L449

Alternatively, if you help us create a minimal repo which reproduces the issue, we can incorporate it as a fixture with test. Maybe on top of `yarn create fusion-app` for example.

**Comment by lxe** [comment]
Yes I can reproduce this generically. I'll add a "unit" test.

File: build/compiler.js [hunk 1]
```diff
@@ -505,6 +505,13 @@ function getConfig({target, env, dir, watch, cover}) {
                 parallel: true, // default from webpack
                 uglifyOptions: {
                   compress: {
+                    // typeofs: true (default) transforms typeof foo == "undefined" into foo === void 0.
+
+                    // This mangles mapbox-gl creating an error when used alongside with window global mangling:
```
[the remark this answers is not in the document]
Comment (lxe) [reply build/compiler.js]: This is generically reproducible... I want to get the static server to run though... see slack :)

File: build/compiler.js [hunk 2]
```diff
@@ -505,6 +505,13 @@ function getConfig({target, env, dir, watch, cover}) {
...
+                    // https://jeng.uberinternal.com/browse/WPT-1185
```
Comment (mlmorg) [inline build/compiler.js]: Please remove this

**Comment by mlmorg** [comment]
Supposedly this was fixed in the latest mapbox-gl: https://github.com/mapbox/mapbox-gl-js/pull/6956. Can we simply stop using the old versions?

**Comment by vicapow** [comment]
Good catch @mlmorg i agree we should just update the dep if this is fixed upstream.

**Comment by lxe** [comment]
> Supposedly this was fixed in the latest mapbox-gl: mapbox/mapbox-gl-js#6956. Can we simply stop using the old versions?

This is still an issue unrelated to mapbox specifically, (but manifests itself there). People will be using older versions of mapbox and fusion in the wild, and it should be able to be compatible with these sort of cases.

Besides, uglification changes like there won't affect the bundle size much

**Pull request merged**